### PR TITLE
feat: add non-root container check

### PR DESCRIPTION
**Pull request by carslen**
Introducing the non-root container check for Dockerfiles in repository (eclipse-tractusx/sig-infra#91).

The check will use the previously added helper function `container.findDockerfilesAt()` function to get all Dockerfiles in repo. Afterwards the Dockerfiles will be parsed for `USER` instruction and validate the provided username/group-name/uid/gid.

Valid usernames follows the `adduser` recommendations, please refer to https://manpages.debian.org/unstable/adduser/adduser.conf.5.en.html#NAME_REGEX for details. 

Go dependencies has been updated and `DEPENDENCIES` file is up to date.

Unhandled circumstance:
- No `USER` instruction is found in Dockerfile.

Decision required if Test should fail or pass in this case.

### CODE REVIEW

File: internal/container/non_root_container_check.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+// validateUser validates USER instruction in Dockerfiles and return a bool if a valid USER has been found.
+// To return true username/group-name must not be root or contain upper case letters, or must not be 0 or greater 65536
+// for uid/gid.
+func validateUser(u *user) bool {
+	if u == nil {
+		return false
+	}
+
+	validUser, _ := regexp.Match(
+		`(^(\d|[1-9]\d{1,3}|[1-5]\d{4}|6[0-4]\d{3}|65[0-4]\d{2}|655[0-2]\d|6553[0-6])$)|(^[a-z_][a-z0-9_-]*[$]?$)`,
```
Comment (almadigabor) [inline internal/container/non_root_container_check.go:81]: The regex pattern could be extracted into a constant to be reusable.

Comment (carslen) [reply internal/container/non_root_container_check.go]: Good idea. Done

File: internal/container/dockerfile.go [hunk 2]
```diff
@@ -95,11 +102,38 @@ func (d *dockerfile) baseImage() string {
...
+	}
+	return u.user == other.user && u.group == other.group
+}
+
 // findDockerfilesAt will search the current repository recursively for Dockerfiles.
 // If a file is found, the relative path to the file is returned in the result slice.
 // If no Dockerfile is found the result will be an empty slice
 func findDockerfilesAt(dir string) []string {
-	fmt.Println("Start finding Dockerfiles at " + dir)
+	//fmt.Println("Start finding Dockerfiles at " + dir)
```
Comment (tomaszbarwicki) [inline internal/container/dockerfile.go:136]: Do we need that if commented?

File: internal/container/dockerfile.go [hunk 3]
```diff
@@ -109,7 +143,11 @@ func findDockerfilesAt(dir string) []string {
 		return nil
 	})
 
-	fmt.Println("Found Dockerfiles: " + strings.Join(foundFiles, ", \n\t"))
+	if len(foundFiles) >= 1 {
```
Comment (tomaszbarwicki) [inline internal/container/dockerfile.go:146]: not sure if we need that statement, nothing happens within it

File: internal/container/dockerfile.go [hunk 4]
```diff
@@ -109,7 +143,11 @@ func findDockerfilesAt(dir string) []string {
...
+		//fmt.Println("Found Dockerfiles:\n\t" + strings.Join(foundFiles, ", \n\t"))
```
Comment (tomaszbarwicki) [inline internal/container/dockerfile.go:147]: not sure if that's commented by purpose or not needed any more

File: internal/container/dockerfile.go (hunk 2 above)
Comment (carslen) [reply internal/container/dockerfile.go]: re-enabled as per team decision.

File: internal/container/dockerfile.go (hunk 3 above)
Comment (carslen) [reply internal/container/dockerfile.go:146]: now it makes sense again

File: internal/container/dockerfile.go (hunk 4 above)
Comment (carslen) [reply internal/container/dockerfile.go]: re-enabled as per team decision.

**Review by almadigabor — APPROVED** [review]
looks good

**Comment by tomaszbarwicki** [comment]
LGTM

**Review by tomaszbarwicki — COMMENTED** [review]
LGTM

**Comment by carslen** [comment]
close this due to messed up commit history. New PR #55

**Pull request closed without merging**